src/decoder/ifd.rs

```diff
@@ -198,9 +198,17 @@ impl Entry {
                 if n > limits.decoding_buffer_size {
                     return Err(TiffError::LimitsExceeded);
                 }
-                decoder.goto_offset(self.r(bo).read_u32()?)?;
-                let string = decoder.read_string(n)?;
-                Ok(Ascii(string))
+                if n <= 4 {
+                    let mut buf = vec![0; n];
```

Comment:
Yeah you are right. But the function `String::from_utf8` take a vec so it will be on the heap anyway no ?
```rust
                if n <= 4 {
                    let mut buf = [0u8; 4];
                    self.r(bo).read_exact(&mut buf )?;
                    let v = String::from_utf8(buf.into())?;
                    let v = v.trim_matches(char::from(0));
                    Ok(Ascii(v.into()))
                } else
                ...
```

Reply:
Stupid me (*facepalm*). Of course, the String will reuse of the allocation in the common case. Then the only advantage would be that the allocation might be moved _after_ the trimming of null bytes, so a string of only nulls would be come empty and wouldn't require one. I'm not sure if that is common enough (or even valid?) to warrant the special case..

Reply:
I can push the code, it will be optimized for the case you mention anyway ;)